javascript/react/props-ii/dealing-with-this-props-children.md

```diff
@@ -31,30 +31,88 @@ aspects:
 **React** provides the user with the useful class `React.Children` which provides various utilities for dealing with the opaque data structure `this.props.children`.
 
 `map` invokes a function on every immediate child within `children`, returning an array:
-```javascript
-array React.Children.map(object children,
-  function fn[, object thisArg]);
+
+```jsx
+React.Children.map(
+  children, 
+  function[(thisArg)]
```

Comment:
What's the point of the `thisArg`? Not sure what this notation means.

Maybe just name the function more accurately?

```suggestion
  functionToCallOnEachChild
```